tensorzero-core/src/clickhouse/mod.rs

```diff
@@ -235,16 +247,42 @@ impl ClickHouseConnectionInfo {
                             message: DisplayOrDebugGateway::new(e).to_string(),
                         })
                     })?;
-                let status = response.status();
+                let status = res.status();
 
-                let response_body = response.text().await.map_err(|e| {
+                // Get the ClickHouse summary info from the headers
+                let metadata = if let Some(summary) = res.headers().get("x-clickhouse-summary") {
+                    let summary_str = summary.to_str().map_err(|e| {
+                        Error::new(ErrorDetails::ClickHouseQuery {
+                            message: format!("Failed to parse x-clickhouse-summary header: {e}"),
+                        })
+                    })?;
+
+                    serde_json::from_str::<ClickHouseResponseMetadata>(summary_str).map_err(
+                        |e| {
+                            Error::new(ErrorDetails::ClickHouseQuery {
+                                message: format!("Failed to deserialize x-clickhouse-summary: {e}"),
+                            })
+                        },
+                    )?
+                } else {
+                    tracing::warn!("No x-clickhouse-summary header found in ClickHouse response");
```

Comment:
Do we expect this to always be present? If so, we might want to error instead of warn when it's missing